src/context/Ad4minContext.tsx

```diff
@@ -167,7 +116,109 @@ export function Ad4minProvider({ children }: any) {
         return null
       })
     }
-  }
+  }, []);
+  
+  const checkIfAgentIsInitialized = useCallback(async (client: Ad4mClient) => {
+    console.log("Check if agent is initialized.", client)
+
+    let status = await client?.agent.status();
+    console.log("agent status in init: ", status);
+
+    handleLogin(client, status.isUnlocked, status.did ? status.did! : "");
+
+    return status;
+  }, [handleLogin]);
+  
+  const Timeout = () => {
+    let controller = new AbortController();
+    setTimeout(() => controller.abort(), 20);
+    return controller;
+  };
+
+  const findAd4mPort = useCallback(async () => {
```

Comment:
We can get get the port passed in ad4m binary with tauri command, no need to find it like this way.
Also we probably don't need to set the range with port start and end.